File: app/src/main/java/ua/com/amicablesoft/android/wr/ui/main/MainView.java
package ua.com.amicablesoft.android.wr.ui.main;

import java.io.File;
import java.io.IOException;
import java.util.ArrayList;

import ua.com.amicablesoft.android.wr.models.Competition;
import ua.com.amicablesoft.android.wr.models.Powerlifter;

/**
 * Created by lapa on 25.09.16.
 */

interface MainView {
    void requestPermissions();
    void recordVideo(File videoFile) throws IOException;

    void setListPowerlifters(ArrayList<Powerlifter> list);
    void setListCompetitions(ArrayList<Competition> listCompetitions);

    void setPowerlifter(int position);
    void setCompetition(int position);
    void setExercise(Integer exercise);

    void showSnackbar(int message);
    void showLoading();
    void dismissLoading();

    void openAddPowerlifterView();
    void openAddCompetitionView();
    void openVideoGallery();
}

File: app/src/main/java/ua/com/amicablesoft/android/wr/ui/main/MainView.kt
/*
 * Created by Olha Tymoshenko on 5/25/18 1:48 PM.
 * Copyright (c) 2018. All rights reserved.
 */

package ua.com.amicablesoft.android.wr.ui.main

import ua.com.amicablesoft.android.wr.models.Competition
import ua.com.amicablesoft.android.wr.models.Powerlifter
import java.io.File
import java.io.IOException
import java.util.*

internal interface MainView {
    fun requestPermissions()
    @Throws(IOException::class)
    fun recordVideo(videoFile: File)

    fun setListPowerlifters(list: ArrayList<Powerlifter>)
    fun setListCompetitions(listCompetitions: ArrayList<Competition>)

    fun setPowerlifter(position: Int)
    fun setCompetition(position: Int)
    fun setExercise(exercise: Int?)

    fun showSnackbar(message: Int)
    fun showLoading()
    fun dismissLoading()

    fun openAddPowerlifterView()
    fun openAddCompetitionView()
    fun openVideoGallery()
}